pkgs/servers/web-apps/outline/default.nix

```diff
@@ -0,0 +1,88 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, makeWrapper
+, nodejs
+, yarn
+, yarn2nix-moretea
+}: let
+  pname = "outline";
+  version = "0.65.2";
+
+  src = fetchFromGitHub {
+    owner = "outline";
+    repo = "outline";
+    rev = "v${version}";
+    sha256 = "sha256-a9K6nMgg1j93BPiy03M86dDecXv/J47vUaqHH3S6DOs=";
+  };
+in stdenv.mkDerivation {
+  inherit pname version src;
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ yarn nodejs ];
+
+  # Replace the inline call to yarn with our sequalize wrapper. This should be
+  # the only occurrence:
+  # https://github.com/outline/outline/search?l=TypeScript&q=yarn
+  patches = [ ./sequelize-command.patch ];
+
+  yarnOfflineCache = yarn2nix-moretea.importOfflineCache ./yarn.nix;
+
+  configurePhase = ''
+    export HOME=$PWD/yarn_home
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+    ${yarn2nix-moretea.fixup_yarn_lock}/bin/fixup_yarn_lock yarn.lock
+
+    yarn install --offline \
+      --frozen-lockfile \
+      --ignore-engines --ignore-scripts
+    patchShebangs node_modules/
+    yarn build
+
+    pushd server
+    cp -r config migrations onboarding ../build/server/
+    popd
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mv node_modules build $out/
+    patchShebangs $out
```

Comment:
Please be specific about the files. Patching **every** file in the node_modules black hole takes a good amount of time.